### PR TITLE
Add Testing Requirements to BankAccounts

### PR DESCRIPTION
**Pull request by droberts-sea**
## Description
Restructure the BankAccounts project to match the TDD setup described in [the textbook](https://github.com/Ada-Developers-Academy/textbook-curriculum/blob/master/00-programming-fundamentals/08-intro-to-automated-tests.md). This means moving project code to `lib/`, and adding a `specs/` directory and a Rakefile to run tests.

Tests for wave 1 are provided in full. For waves 2 and 3, I've provided english descriptions (as in `it "Requires an initial balance of at least $10"`), but the students will have to fill in the tests themselves.

`README.md` has been updated to reflect the changes to the project setup, and now contains some requirements around testing.

You need an extra gem to use `xdescribe` to skip spec tests. Installing this has been added to the README setup steps.

My theory is that adding the tests will clarify project requirements, and the students should be testing things in the REPL anyway so I don't *think* it will take that much more time to fill in (or copy-paste and slightly modify) tests. So if that holds, then we should be able to get through it in the same amount of time. It'll be a grand experiment!

## Still TODO
- [x] Add explicit testing requirements to `feedback.md` once it's been added

## Feedback Requested
Did I miss any test cases?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,23 +4,34 @@ We will be working with the concept of bank accounts in order to explore more ob
...
-1. Clone the project master repo and create a new branch with your initials.
-1. Push your branch so it will show in the list of branches on the project master.
+1. This is an individual, stage 1 project.
 1. Fork the project master.
 1. Clone the forked repo: `$ git clone [YOUR FORKED REPO URL]`
-1. Switch to your branch by doing `git checkout [YOUR BRANCH NAME]`. *Do not work on the master branch*.
 1. `cd` into the dir create:d `$ cd BankAccounts`
-1. Run `git status` to verify the branch you are on
 1. Run `git remote -v` to verify the folder you are in corresponds to the fork you have created.
+1. Run `gem install minitest-skip` to install an extra gem for testing (more on what this actually does later).
```
Comment (CheezItMan) [inline README.md]: So you want them to learn how to mark skipped tests with `xit` and `xdescribe` like in Jasmine?

File: specs/account_spec.rb [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+  end
+
+  describe "Account.find" do
+    it "Returns an account that exists" do
+      # TODO: Your test code here!
+    end
+
+    it "Raises an error for an account that doesn't exist" do
+      # TODO: Your test code here!
+    end
```
Comment (CheezItMan) [inline specs/account_spec.rb:169]: Maybe something looking for the 1st and last accounts in the file, or maybe we should let them figure that part out.

**Review by CheezItMan — APPROVED** [review]
Looks pretty good, I like how the tests are provided in wave 1 and outlined in waves 2-3.

File: README.md (hunk 1 above)
Comment (droberts-sea) [reply README.md]: I don't think you can skip at the `describe` level otherwise.

File: specs/account_spec.rb (hunk 2 above)
Comment (droberts-sea) [reply specs/account_spec.rb:169]: Probably better to be explicit.

Comment (droberts-sea) [reply specs/account_spec.rb:169]: Added.

**Pull request merged**